app/models/game.rb

```diff
@@ -23,31 +22,11 @@ def populate_board!
     end
   end
 
-  def check?(color)
-    if color == 'white'
-      king_x = @white_king.horizontal_position
-      king_y = @white_king.vertical_position
-    else
-      king_x = @black_king.horizontal_position
-      king_y = @black_king.vertical_position
-    end
-    Piece.where(!color).each do |piece|
-      return true if piece.valid_move?(king_x, king_y)
+  def check?
+    king = King.where(game: self, color: color)
+    Piece.where(color: !king.color).each do |piece|
```

Comment:
We need to specify which game we want pieces for here, too. Also, unfortunately !king.color will always return false, so one solution is to use ActiveRecord to build the query: 
Piece.where(game: self).where.not(color: color)